lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -306,31 +306,35 @@ def from_string(cls, collection_input, artifacts_manager, supplemental_signature
         return cls.from_requirement_dict(req, artifacts_manager)
 
     @classmethod
-    def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_options=True):
-        req_name = collection_req.get('name', None)
-        req_version = collection_req.get('version', '*')
-        req_type = collection_req.get('type')
+    def from_requirement_dict(cls, collection_req: dict, art_mgr: ConcreteArtifactsManager, validate_signature_options: bool = True):
+        DEFAULT_VERSION = '*'
```

Comment:
This variable name is misleading. `*` means “any”. There's no default I can think of.
```suggestion
        ANY_VERSION = '*'
```

Also, if this is being made into an upper-case constant, it's probably best to move it to the top of the module as `_ANY_VERSION`.